ios/RNCardVerify.m

```diff
@@ -19,39 +19,70 @@ - (void)dismissView {
+                       @"expiryYear": creditCard.expiryYear ?: [NSNull null],
+                       @"payloadVersion": @"1",
+                       @"verificationPayload": encryptedPayload ?: [NSNull null]
+                    }});
+}
 
-    }
+- (void)userCanceledVerifyCardWithViewController:(VerifyCardViewController * _Nonnull)viewController  API_AVAILABLE(ios(11.2)){
+    [self dismissView];
+    self.resolve(@{ @"action": @"canceled",
+                    @"canceledReason": @"user_canceled"});
+}
+@end
 
-    - (void)userDidScanCardWithViewController:(VerifyCardSimpleViewController * _Nonnull)viewController number:(NSString * _Nonnull)number name:(NSString * _Nullable)name expiryYear:(NSString * _Nullable)expiryYear expiryMonth:(NSString * _Nullable)expiryMonth payload:(NSString * _Nullable)payload  API_AVAILABLE(ios(11.2)){
-        [self dismissView];
-        self.resolve(@{@"action" : @"scanned",
-                       @"payload": @{
-                           @"number": number,
-                           @"cardholderName": name ?: [NSNull null],
-                           @"expiryMonth": expiryMonth ?: [NSNull null],
-                           @"expiryYear": expiryYear ?: [NSNull null],
-                           @"payloadVersion": @"1",
-                           @"verificationPayload": payload ?: [NSNull null]
-                       }
-                   });
-    }
+//MARK: - Verify Card Add Delegate Implementation
+@implementation VerifyCardAddDelegate
 
-    - (void)userMissingCardWithViewController:(VerifyCardSimpleViewController * _Nonnull)viewController  API_AVAILABLE(ios(11.2)){
-        [self dismissView];
-        self.resolve(@{ @"action": @"canceled",
-                        @"canceledReason": @"user_missing_card"
-                     });
-    }
+- (void)setCallback:(RCTPromiseResolveBlock)resolve {
+    self.resolve = resolve;
+}
+
+- (void)dismissView {
+    UIViewController *topViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
 
-    - (void)userDidSkipWithViewController:(VerifyCardSimpleViewController * _Nonnull)viewController {
-        [self dismissView];
-        self.resolve(@{ @"action": @"skipped" });
+    while (topViewController.presentedViewController) {
+        topViewController = topViewController.presentedViewController;
     }
+    
+    [topViewController dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (void)userDidCancelCardAdd:(UIViewController * _Nonnull)viewController API_AVAILABLE(ios(11.2)){
+    [self dismissView];
+    self.resolve(@{ @"action": @"canceled",
+                    @"canceledReason": @"user_canceled"});
+}
+
+- (void)userDidPressManualCardAdd:(UIViewController * _Nonnull)viewController API_AVAILABLE(ios(11.2)){
+    [self dismissView];
+    self.resolve(@{ @"action": @"skipped" });
+}
+
+- (void)userDidScanCardAdd:(UIViewController * _Nonnull)viewController creditCard:(CreditCard * _Nonnull)creditCard API_AVAILABLE(ios(11.2)){
```

Comment:
@awushensky I left this function empty since this doesn't provide the encrypted payload and `fraudModelResult` provides both credit card info and the payload.